### PR TITLE
Add static constexpr `dimensions` member to all range/id-like types

### PR DESCRIPTION
**Pull request by psalz**
Partially addresses #270.

### CODE REVIEW

**Comment by gmlueck** [comment]
If we want to put this into SYCL 2020, we should also update the CTS to check it.

**Comment by keryell** [comment]
I guess we should bike-shed this. Is this for SYCL Next?

**Comment by psalz** [comment]
> If we want to put this into SYCL 2020, we should also update the CTS to check it.

That's what the label is for! I will create an issue in the CTS repo if this is merged.

**Review by nliber — CHANGES_REQUESTED** [review]

File: adoc/headers/item.h [hunk 1]
```diff
@@ -4,6 +4,8 @@
 namespace sycl {
 template <int Dimensions = 1, bool WithOffset = true> class item {
  public:
+  static constexpr int dimensions = Dimensions;
```
[the remark this answers is not in the document]
Comment (gmlueck) [reply adoc/headers/item.h:7]: Note, however, that parallel_for offsets are a deprecated feature.  I'm not sure we want to add new APIs for deprecated features.

[the remark this answers is not in the document]
Comment (nliber) [reply adoc/headers/item.h:7]: Ah, true.  Will approve as is.

**Review by nliber — APPROVED** [review]

**Review by keryell — APPROVED** [review]
I am just discovering https://en.cppreference.com/w/cpp/types/rank which is not yet customizable anyway...
So, let's go for this and bike-shed in the future with some alternative to match `mdspan` and so, for example exposing a `static constexpr rank()` member function.

**Comment by fraggamuffin** [comment]
LGTM

**Pull request merged**